File: src/main/java/com/aladdinworksfourfourfour/service/CurrentSensorService.java
package com.aladdinworksfourfourfour.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworksfourfourfour.domain.CurrentSensor;
import com.aladdinworksfourfourfour.dto.CurrentSensorDTO;
import com.aladdinworksfourfourfour.dto.CurrentSensorSearchDTO;
import com.aladdinworksfourfourfour.dto.CurrentSensorPageDTO;
import com.aladdinworksfourfourfour.dto.CurrentSensorConvertCriteriaDTO;
import com.aladdinworksfourfourfour.service.GenericService;
import com.aladdinworksfourfourfour.dto.common.RequestDTO;
import com.aladdinworksfourfourfour.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CurrentSensorService extends GenericService<CurrentSensor, Integer> {

	List<CurrentSensor> findAll();

	ResultDTO addCurrentSensor(CurrentSensorDTO currentSensorDTO, RequestDTO requestDTO);

	ResultDTO updateCurrentSensor(CurrentSensorDTO currentSensorDTO, RequestDTO requestDTO);

    Page<CurrentSensor> getAllCurrentSensors(Pageable pageable);

    Page<CurrentSensor> getAllCurrentSensors(Specification<CurrentSensor> spec, Pageable pageable);

	ResponseEntity<CurrentSensorPageDTO> getCurrentSensors(CurrentSensorSearchDTO currentSensorSearchDTO);
	
	List<CurrentSensorDTO> convertCurrentSensorsToCurrentSensorDTOs(List<CurrentSensor> currentSensors, CurrentSensorConvertCriteriaDTO convertCriteria);

	CurrentSensorDTO getCurrentSensorDTOById(Integer currentSensorId);







}

File: aladdinworksfourfourfour_ui/src/services/CurrentSensorService.js
import http from "../http-common"; 

class CurrentSensorService {
  getAllCurrentSensors(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/currentSensor/currentSensors`, searchDTO);
  }

  get(currentSensorId) {
    return this.getRequest(`/currentSensor/${currentSensorId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/currentSensor?field=${matchData}`, null);
  }

  addCurrentSensor(data) {
    return http.post("/currentSensor/addCurrentSensor", data);
  }

  update(data) {
  	return http.post("/currentSensor/updateCurrentSensor", data);
  }
  
  uploadImage(data,currentSensorId) {
  	return http.postForm("/currentSensor/uploadImage/"+currentSensorId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CurrentSensorService();
